### ISSUE TITLE
Inconsistent API user permissions between v1.3 and v1.4-alpha versions causing upgrade failure

### ISSUE THREAD

**Issue by vponomaryov** [scylladb/scylla-operator#677]
**Describe the bug**
Operator upgrade on GKE from v1.3.0 to v1.4.0-alpha.0-45-g5adf54a fails with the following error:

```
I0702 21:02:23.730805       1 operator/operator.go:133] operator version v1.4.0-alpha.0-45-g5adf54a
I0702 21:02:23.730861       1 operator/operator.go:134] loglevel is set to "2"
I0702 21:02:23.731211       1 leaderelection/leaderelection.go:94] Starting leader election
I0702 21:02:23.731231       1 leaderelection/leaderelection.go:243] attempting to acquire leader lease scylla-operator/operator-locks...
I0702 21:02:23.749724       1 leaderelection/leaderelection.go:253] successfully acquired lease scylla-operator/operator-locks
I0702 21:02:23.750600       1 orphanedpv/controller.go:163] "Starting controller" controller="OrphanedPV"
I0702 21:02:23.750663       1 cache/shared_informer.go:240] Waiting for caches to sync for OrphanedPVController
I0702 21:02:23.750739       1 cache/reflector.go:219] Starting reflector *v1.ScyllaCluster (12h0m0s) from k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167
I0702 21:02:23.750758       1 cache/reflector.go:219] Starting reflector *v1.Secret (12h0m0s) from k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167
I0702 21:02:23.751001       1 scyllacluster/controller.go:203] "Starting controller" controller="ScyllaCluster"
I0702 21:02:23.751012       1 cache/shared_informer.go:240] Waiting for caches to sync for ScyllaClusterController
I0702 21:02:23.751086       1 cache/reflector.go:219] Starting reflector *v1.Node (12h0m0s) from k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167
I0702 21:02:23.751115       1 cache/reflector.go:219] Starting reflector *v1.Pod (12h0m0s) from k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167
I0702 21:02:23.751240       1 cache/reflector.go:219] Starting reflector *v1.Service (12h0m0s) from k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167
I0702 21:02:23.751390       1 cache/reflector.go:219] Starting reflector *v1.StatefulSet (12h0m0s) from k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167
I0702 21:02:23.751504       1 cache/reflector.go:219] Starting reflector *v1beta1.PodDisruptionBudget (12h0m0s) from k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167
I0702 21:02:23.751661       1 cache/reflector.go:219] Starting reflector *v1.PersistentVolume (12h0m0s) from k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167
I0702 21:02:23.751748       1 cache/reflector.go:219] Starting reflector *v1.PersistentVolumeClaim (12h0m0s) from k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167
I0702 21:02:23.851040       1 cache/shared_informer.go:247] Caches are synced for OrphanedPVController 
I0702 21:02:23.952107       1 cache/shared_informer.go:247] Caches are synced for ScyllaClusterController 
I0702 21:02:23.952436       1 controllertools/secretmanager.go:106] "Patching Secret to remove its controllerRef" Secret="scylla/sct-cluster-auth-token" ControllerRef="scylla.scylladb.com/v1/ScyllaCluster:sct-cluster"
E0702 21:02:23.954812       1 scyllacluster/controller.go:187] syncing key 'scylla/sct-cluster' failed: secrets "sct-cluster-auth-token" is forbidden: User "system:serviceaccount:scylla-operator:scylla-operator" cannot patch resource "secrets" in API group "" in the namespace "scylla"
I0702 21:02:23.960170       1 controllertools/secretmanager.go:106] "Patching Secret to remove its controllerRef" Secret="scylla/sct-cluster-auth-token" ControllerRef="scylla.scylladb.com/v1/ScyllaCluster:sct-cluster"

...
```

**To Reproduce**
1. Deploy operator using v1.3.0 Helm chart on GKE
2. Upgrade operator using v1.4.0-alpha.0-45-g5adf54a Helm chart
3. See error

**Expected behavior**
No errors appear, rolling operator upgrade happens.

**Logs**
Please, provide `kubectl get events`, `kubectl logs -n scylla pod-name` and other relevant information.
[kubernetes-72fc7731.tar.gz](https://github.com/scylladb/scylla-operator/files/6764439/kubernetes-72fc7731.tar.gz)

**Environment:**
 - Platform: GKE
 - Kubernetes version: v1.19.11-gke.2100
 - Scylla version: 4.4.3
 - Scylla-operator version: v1.4.0-alpha.0-45-g5adf54a

**Additional context**

**Comment by zimnx** [comment]
Will be fixed by #673

**Comment by zimnx** [comment]
Fixed by #673
